gemrb/core/Inventory.cpp

```diff
@@ -196,6 +196,8 @@ void Inventory::CalculateWeight() const
 		}
 	}
 	Changed = false;
+
+	Owner->SetStat(IE_ENCUMBRANCE, Weight, false);
```

Comment:
This needs to be changed to SetBase, so we set the base stat. SetStat sets the modified one, which is wiped each game tick. This is probably why you couldn't just delete the calls from GUIScript (which is the way to go).

Reply:
I added a commit changing it SetBase, but I'm not sure if it's necessary.
I think I was misunderstanding when CalculateWeight should be called (immediately before `GetWeight()`, gated by `Changed`, rather than on inventory update).
I also removed a few calls to CalculateWeight within other inventory functions to keep them consistent, since I couldn't find anywhere that the weight was accessed without first being recalculated.

Reply:
It feels a bit wrong — shouldn't be up to the inventory to keep track of its own state? Division of responsibilities and all that. Now it works only because the recalc is called each tick, so any changes are quickly picked up and there's at most a tick's delay.

Reply:
SetBase is needed even more than I thought too — if the inventory didn't change, nothing will set the stat that tick, so it will equal the base ... which is unset, so 0.

Reply:
I forgot to add, maybe the right approach would be to call CalculateWeight from GetWeight?
That would ensure you don't get a stale weight, while encapsulating things better. Maybe `Weight` should also be private in that case.

Reply:
Weight should definitely remain private — for anyone that needs it, there's a getter already. CalculateWeight is a bit annoying, since it also does some flag cleanup, but it also rechecks everything (we don't update the weight atomically), which saves us having to deal with both the removed and new item if a swap occurred. I don't think it's worth touching it.

But I do think it's better to keep GetWeight const and make the effort to maintain an always correct Weight. `Changed` is there only for the weight handling, so it's easy to see where the updates should happen. Not sure why these two are marked mutable either.

Reply:
I guess I see two paths for a clean interface:
- Use `Weight` to privately cache the value, invalidate using `Changed`, interface using `GetWeight`
- Remove `Changed`, replace Changed=True with calls to `CalculateWeight`. Less performant, but I doubt it'll make a difference.